### PR TITLE
Auth fix

### LINKED ISSUE #105: [Vapor3] Auth password not applied
I just tried to use a redis password and I suspect the auth command never gets executed
https://github.com/vapor/redis/blob/master/Sources/Redis/Database/RedisDatabase.swift#L27

The auth command future is created, but (if I understand event loops correctly) never executed

### LINKED ISSUE #107: ERR Protocol error: expected '$', got '+' during tests
I've been trying to run the tests in a server that requires auth (Heroku specifically) but couldn't get them to pass:

> [RedisError.serverSide: ERR Protocol error: expected '$', got '+'] [RedisDataDecoder.swift:50:94]

Could it be related to #105  ?

The only thing I modified was the `makeTest` function:

```swift
static func makeTest() throws -> RedisClient {
    let group = MultiThreadedEventLoopGroup(numThreads: 1)
    let db = try RedisDatabase(url: URL(string: "redis://h:password_here@ec2-000.compute-1.amazonaws.com:53439")!)
    return try db.makeConnection(on: group).wait()
}
```

### Repro:

1. Clone the project
2. Go to Heroku and add any (or both) of the Redis Services mentioned below
3. Copy their connection strings
4. Replace `makeTest()` with the code above, adding the proper connection string
5. Run the tests (they won't finish)

### Expected:

Tests should pass

### Actual results:

Tests will freeze / not pass

Env:
> Vapor Redis: 3.0.0-rc.2.4
> OS: macOS High Sierra 10.13.4
> Redis Services: [Heroku Redis](https://elements.heroku.com/addons/heroku-redis) & [Redis To Go](https://elements.heroku.com/addons/redistogo)

### PR DESCRIPTION
**Pull request by pedantix**
- Modified Test Suite to make use of redis_password if environment makes use of it
- Modified CircleCI file to use an auth environment
- Modified Test Suite to use @MrMage's revelation of `waitForExpectations`
- Cleaned up some whitespace to enable swiftlint strict to run again

- resolve #105 
- resolve #107

### CODE REVIEW

File: Tests/RedisTests/RedisTests.swift [hunk 1]
```diff
@@ -26,6 +33,8 @@ class RedisTests: XCTestCase {
     }
 
     func testPubSubSingleChannel() throws {
+        let futureExpectation = expectation(description: "Subscriber should recevie message")
```
Comment (bre7) [inline Tests/RedisTests/RedisTests.swift]: Minor typo :)

Comment (pedantix) [reply Tests/RedisTests/RedisTests.swift]: super good catch! thanks!

**Comment by bre7** [comment]
The issue was related to how the AUTH command was being encoded right ?

**Comment by pedantix** [comment]
@bre7 I am not 100% sure. The real issue is that we had an untested feature in code base ><. By adding `authlinux` build to the circle.yml file I had a means of debugging redis auth without doing heroic measures, after I could produce the "you aren't authed error" I was able to quickly fix the issue. Thanks to @John-Connolly 's Auth command addition I don't actually know if it was an implementation detail I just moved things around with some jiggerty pockety and ensured it was tested. But I think you are right the password was encoded in the wrong string format.

**Comment by bre7** [comment]
Do I really need to accept the review or can you just merge (seeing as you do have write access) ?

**Comment by vzsg** [comment]
Having write access is not enough, as the master branch is protected. Only administrators can merge without an accepted review. So yeah, instead of a comment, you should leave a positive review, or wait for @tanner0101.

**Comment by penny-coin** [comment]
Hey @pedantix, you just merged a pull request, have a coin! 

You now have 2 coins.

**Comment by pedantix** [comment]
@tanner0101 everytime I think I have swiftlint working again ... :D

**Pull request merged**